application/src/main/java/org/togetherjava/tjbot/commands/filesharing/ShareLongAttachmentsMessageListener.java

```diff
@@ -0,0 +1,174 @@
+    private static final HttpClient CLIENT = HttpClient.newHttpClient();
+
+    private final Predicate<String> isStagingChannelName;
+    private final Predicate<String> isOverviewChannelName;
+
+    public ShareLongAttachmentsMessageListener(@NotNull Config config) {
+        super(Pattern.compile(".*"));
+
+        isStagingChannelName = Pattern.compile(config.getHelpSystem().getStagingChannelPattern())
+            .asMatchPredicate();
+        isOverviewChannelName = Pattern.compile(config.getHelpSystem().getOverviewChannelPattern())
+            .asMatchPredicate();
+    }
+
+    @Override
+    public void onMessageReceived(@NotNull MessageReceivedEvent event) {
+        if (event.getAuthor().isBot() || event.isWebhookMessage()) {
+            return;
+        }
+
+        if (!isHelpThread(event)) {
+            return;
+        }
+
+        event.getMessage()
+            .getAttachments()
+            .stream()
+            .filter(attachment -> extensions.contains(attachment.getFileExtension()))
+            .forEach(attachment -> processAttachment(event, attachment));
+    }
+
+    private void processAttachment(MessageReceivedEvent event, Message.Attachment attachment) {
+        attachment.retrieveInputStream()
+            .thenApply(this::readAttachment)
+            .thenApply(content -> uploadToSharingService(event, attachment, content))
+            .thenAccept(response -> sendMessage(event, attachment, response));
+    }
+
+    private String readAttachment(@NotNull InputStream stream) {
+        try {
+            byte[] bytes = stream.readAllBytes();
+            return new String(bytes, StandardCharsets.UTF_8);
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    private HttpResponse<String> uploadToSharingService(MessageReceivedEvent event,
+            Message.Attachment attachment, String content) {
+        GistFile file = new GistFile();
+        file.setContent(content);
+
+        String parsedContent;
+        try {
+            parsedContent = JSON.writeValueAsString(file);
+        } catch (JsonProcessingException e) {
+            throw new IllegalStateException("Couldn't parse content!", e);
+        }
+
+        String fileName = attachment.getFileName();
+        String fileExtension = attachment.getFileExtension();
+
+        if (fileExtension == null || fileExtension.equals("txt")) {
+            fileExtension = "java";
+        } else if (fileExtension.equals("fxml")) {
+            fileExtension = "xml";
+        }
+
+        fileName = fileName.substring(0, fileName.lastIndexOf(".")) + "." + fileExtension;
+
+        String body = """
+                {
+                  "description": "%s",
+                  "public": false,
+                  "files": {
+                      "%s": %s
+                  }
+                }""".formatted(event.getAuthor(), fileName, parsedContent);
```

Comment:
This part needs a comment why you did not went for jackson fully (i.e. the fact that the file names are dynamic and hence parts of the request are manually created)

Also, you miss the url encoding on the `fileName` and `author` name. So if any of those contains bad characters, your request will mess up again.

Reply:
do I need to make another jackson like class for fileName and author to parse the variables? also what should I name it?